### ISSUE TITLE
Memory profiling keeps on running forever while only cpu profiling doesn't

### ISSUE THREAD

**Issue by edomerli** [emeryberger/scalene#66]
Hello, this is the first issue I'm writing ever, so I would appreciate any critic about my writing and problem displaying style.

I'm on Linux Mint 9, I have just installed scalene using `pip3 install scalene`, created an easy python script called target.py:

```python
import numpy as np

def main():
    x = np.array(range(10 ** 7))
    y = np.array(np.random.uniform(0, 100, size=(10 ** 8)))

main() 
```

Now, if I run: `scalene target.py --cpu-only` or `python3 -m scalene target.py --cpu-only` I get the following:

```
                                         target.py: % of CPU time = 100.00% out of   4.38s.                                         
       ╷        ╷        ╷                                                                                                          
  Line │CPU %   │CPU %   │                                                                                                          
       │Python  │native  │target.py                                                                                                 
╺━━━━━━┿━━━━━━━━┿━━━━━━━━┿━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸
     1 │   7.1% │  17.2% │import numpy as np                                                                                        
     2 │        │        │                                                                                                          
     3 │        │        │def main():                                                                                               
     4 │   0.2% │  38.0% │    x = np.array(range(10**7))                                                                            
     5 │   0.5% │  37.0% │    y = np.array(np.random.uniform(0, 100, size=(10**8)))                                                 
     6 │        │        │                                                                                                          
     7 │        │        │main()                                                                                                    
```

Whether if I run `scalene target.py ` or `python3 -m scalene target.py` (that is, cpu AND memory profiling), the console doesn't dispay anything and keeps "waiting", I can type stuff and it would display but it the process would keep running until I stop it with a keyboard interrupt Ctrl + C and get this:

```
^CTraceback (most recent call last):
  File "/home/edo/.local/bin/scalene", line 5, in <module>
    from scalene.__main__ import main
  File "/home/edo/.local/lib/python3.6/site-packages/scalene/__main__.py", line 1, in <module>
    from scalene import scalene
  File "/home/edo/.local/lib/python3.6/site-packages/scalene/scalene.py", line 261, in <module>
    result = subprocess.run(args)
  File "/usr/lib/python3.6/subprocess.py", line 425, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/usr/lib/python3.6/subprocess.py", line 855, in communicate
    self.wait()
  File "/usr/lib/python3.6/subprocess.py", line 1477, in wait
    (pid, sts) = self._try_wait(0)
  File "/usr/lib/python3.6/subprocess.py", line 1424, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
```

**Comment by emeryberger** [comment]
Linux Mint 9 was released in 2010; I have no plans to support systems from so long ago. I will test this with a recent version and see what's going on. In the meanwhile, can you tell me what version of Scalene you have installed?

**Comment by emeryberger** [comment]
I was able to reproduce your issue under Linux Mint 20. Right now, there's an issue for call stack tracing that doesn't appear with Ubuntu (this is used to separate out attribution of Python vs. C memory allocation). I'm disabling it temporarily for Linux (still ok for Mac OS X) while I figure out a solution. I just ran the latest version of Scalene with your code (and other tests) and it works fine, though it currently attributes all memory allocation to Python. Hope to have a fix for that out soon!
